tests/cloud_functions/testing_utils.py

```
@@ -0,0 +1,224 @@
+        return self._docs.get(doc_id, InMemoryFirestoreDoc())
+
+
+class InMemoryFirestoreClient:
+    """A class that simulates a Firestore client in memory.
+
+    Attributes:
+        _collections (defaultdict): A dictionary of collections, where each collection
+            is an instance of InMemoryFirestoreCollection.
+    """
+
+    def __init__(self):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)